### ISSUE TITLE
Add better learning rate annealing scheduler + Learning rate finder

### ISSUE THREAD

**Issue by mhham** [zalandoresearch/flair#228]
Two ideas that could allow improving training :

1. Add learning rate finder : as explained in [here](https://www.jeremyjordan.me/nn-learning-rate/), and implemented [here](https://github.com/fastai/fastai/blob/98588ff0d088041259ee77ca63cdf0cc117c3bf1/fastai/train.py#L22)

2. Cosine Annealing with restart : https://github.com/roveo/pytorch/pull/1/commits/35891c46dbdec46e55a5fb725418c2880f631dac

**Comment by alanakbik** [comment]
That sounds like a great idea - if fact we've been recently thinking a lot about better methods to find good parameters than a full parameter sweep. Would you be interested in testing/integrating such approaches?

**Comment by alanakbik** [comment]
@kashif what do you think?
